### PR TITLE
Upgrade database and telemetry docker to stretch build

### PR DESCRIPTION
**Pull request by jipanyang**
Signed-off-by: Jipan Yang <jipan.yang@alibaba-inc.com>

**- What I did**
Move database and telemetry docker to stretch based.

**- How I did it**

**- How to verify it**

Docker moved to stretch build.
```
jipan@sonicvm1:~/warm_reboot/sonic-buildimage$ make list
+++ --- Making list --- +++
BLDENV=stretch make -f Makefile.work stretch
make[1]: Entering directory '/home/jipan/warm_reboot/sonic-buildimage'
SONiC Build System

Build Configuration
"CONFIGURED_PLATFORM"             : "broadcom"
"SONIC_CONFIG_PRINT_DEPENDENCIES" : ""
"SONIC_BUILD_JOBS"                : "8"
"SONIC_CONFIG_MAKE_JOBS"          : "8"
"USERNAME"                        : "admin"
"PASSWORD"                        : "admin"
"ENABLE_DHCP_GRAPH_SERVICE"       : ""
"SHUTDOWN_BGP_ON_START"           : ""
"ENABLE_PFCWD_ON_START"           : ""
"INSTALL_DEBUG_TOOLS"             : ""
"ROUTING_STACK"                   : "frr"
"FRR_USER_UID"                    : "300"
"FRR_USER_GID"                    : "300"
"ENABLE_SYNCD_RPC"                : ""
"ENABLE_ORGANIZATION_EXTENSIONS"  : "y"
"HTTP_PROXY"                      : ""
"HTTPS_PROXY"                     : ""
"ENABLE_SYSTEM_TELEMETRY"         : "y"
"SONIC_DEBUGGING_ON"              : "y"
"SONIC_PROFILING_ON"              : ""
"KERNEL_PROCURE_METHOD"           : "build"
"BUILD_TIMESTAMP"                 : "20190208.022025"
"BLDENV"                          : "stretch"

[ 01 ] [ target/docker-sonic-telemetry.gz ]
[ 02 ] [ target/docker-database.gz ]
```

And the stretch version of debian packages built 
```
jipan@sonicvm1:~/warm_reboot/sonic-buildimage$ ls -l target/debs/stretch/redis-*
-rw-r--r-- 1 jipan jipan  18178 Feb  8 02:20 target/debs/stretch/redis-sentinel_3.2.4-1~bpo8+1_amd64.deb
-rw-r--r-- 1 jipan jipan 407152 Feb  8 02:20 target/debs/stretch/redis-server_3.2.4-1~bpo8+1_amd64.deb
-rw-r--r-- 1 jipan jipan 462362 Feb  8 02:20 target/debs/stretch/redis-tools_3.2.4-1~bpo8+1_amd64.deb
-rw-r--r-- 1 jipan jipan    329 Feb  8 02:20 target/debs/stretch/redis-tools_3.2.4-1~bpo8+1_amd64.deb-install.log
-rw-r--r-- 1 jipan jipan  49800 Feb  8 02:20 target/debs/stretch/redis-tools_3.2.4-1~bpo8+1_amd64.deb.log
jipan@sonicvm1:~/warm_reboot/sonic-buildimage$ ls -l target/debs/stretch/sonic-telemetry_0.1_amd64*
-rw-r--r-- 1 jipan jipan 10973960 Feb  8 01:28 target/debs/stretch/sonic-telemetry_0.1_amd64.deb
-rw-r--r-- 1 jipan jipan    12122 Feb  8 01:28 target/debs/stretch/sonic-telemetry_0.1_amd64.deb.log

```

system running as usual
```
root@sonic:/home/admin# docker ps
CONTAINER ID        IMAGE                             COMMAND                  CREATED             STATUS              PORTS               NAMES
fa593f0895e0        docker-snmp-sv2:latest            "/usr/bin/supervisord"   About an hour ago   Up About an hour                        snmp
3cffc58fcbd3        docker-syncd-brcm:latest          "/usr/bin/supervisord"   About an hour ago   Up About an hour                        syncd
743c65c606da        docker-orchagent-brcm:latest      "/usr/bin/supervisord"   About an hour ago   Up About an hour                        swss
bfeba7138bba        docker-sonic-telemetry:latest     "/usr/bin/supervisord"   About an hour ago   Up About an hour                        telemetry
2464ad8582d7        docker-router-advertiser:latest   "/usr/bin/supervisord"   About an hour ago   Up About an hour                        radv
388594cb00ee        docker-dhcp-relay:latest          "/usr/bin/docker_ini…"   About an hour ago   Up About an hour                        dhcp_relay
813b51408a19        docker-lldp-sv2:latest            "/usr/bin/supervisord"   About an hour ago   Up About an hour                        lldp
9e502708ce00        docker-platform-monitor:latest    "/usr/bin/supervisord"   About an hour ago   Up About an hour                        pmon
9037db19e631        docker-teamd:latest               "/usr/bin/supervisord"   About an hour ago   Up About an hour                        teamd
9310824cd940        docker-fpm-frr:latest             "/bin/sh -c '/usr/bi…"   About an hour ago   Up About an hour                        bgp
c3f3e03ff623        docker-database:latest            "/usr/bin/supervisord"   About an hour ago   Up About an hour                        database
root@sonic:/home/admin# docker exec -it database ps -ax
  PID TTY      STAT   TIME COMMAND
    1 pts/0    Ss+    0:01 /usr/bin/python /usr/bin/supervisord
   20 pts/0    Sl     0:00 /usr/sbin/rsyslogd -n
   21 pts/0    Sl     2:09 /usr/bin/redis-server 127.0.0.1:6379
  135 pts/1    Rs+    0:00 ps -ax
root@sonic:/home/admin# 
root@sonic:/home/admin# redis-cli info
# Server
redis_version:3.2.4
redis_git_sha1:5f622506
redis_git_dirty:1
redis_build_id:a3d993f9a75239dd
redis_mode:standalone
os:Linux 4.9.0-8-2-amd64 x86_64
arch_bits:64
multiplexing_api:epoll
gcc_version:6.3.0
process_id:21
run_id:f7141403072e8e504d13ebd76fa08abad58a57c2
tcp_port:6379
uptime_in_seconds:3655
uptime_in_days:0
hz:10
lru_clock:6117789
executable:/usr/bin/redis-server
config_file:/etc/redis/redis.conf

# Clients
connected_clients:408
client_longest_output_list:0
client_biggest_input_buf:0
blocked_clients:0

# Memory
used_memory:13613456
used_memory_human:12.98M
used_memory_rss:11575296
used_memory_rss_human:11.04M
used_memory_peak:14017720
used_memory_peak_human:13.37M
total_system_memory:8342999040
total_system_memory_human:7.77G
used_memory_lua:124928
used_memory_lua_human:122.00K
maxmemory:0
maxmemory_human:0B
maxmemory_policy:noeviction
mem_fragmentation_ratio:0.85
mem_allocator:jemalloc-3.6.0

# Persistence
loading:0
rdb_changes_since_last_save:2034869
rdb_bgsave_in_progress:0
rdb_last_save_time:1549618006
rdb_last_bgsave_status:ok
rdb_last_bgsave_time_sec:-1
rdb_current_bgsave_time_sec:-1
aof_enabled:0
aof_rewrite_in_progress:0
aof_rewrite_scheduled:0
aof_last_rewrite_time_sec:-1
aof_current_rewrite_time_sec:-1
aof_last_bgrewrite_status:ok
aof_last_write_status:ok

# Stats
total_connections_received:8902
total_commands_processed:1815142
instantaneous_ops_per_sec:2273
total_net_input_bytes:461002475
total_net_output_bytes:6967473001
instantaneous_input_kbps:223.38
instantaneous_output_kbps:4345.19
rejected_connections:0
sync_full:0
sync_partial_ok:0
sync_partial_err:0
expired_keys:0
evicted_keys:0
keyspace_hits:1027873
keyspace_misses:41842
pubsub_channels:79
pubsub_patterns:49
latest_fork_usec:0
migrate_cached_sockets:0

# Replication
role:master
connected_slaves:0
master_repl_offset:0
repl_backlog_active:0
repl_backlog_size:1048576
repl_backlog_first_byte_offset:0
repl_backlog_histlen:0

# CPU
used_cpu_sys:63.47
used_cpu_user:67.10
used_cpu_sys_children:0.00
used_cpu_user_children:0.00

# Cluster
cluster_enabled:0

# Keyspace
db0:keys=149,expires=0,avg_ttl=0
db1:keys=3134,expires=0,avg_ttl=0
db2:keys=2666,expires=0,avg_ttl=0
db3:keys=52,expires=0,avg_ttl=0
db4:keys=168,expires=0,avg_ttl=0
db5:keys=2651,expires=0,avg_ttl=0
db6:keys=125,expires=0,avg_ttl=0
root@sonic:/home/admin# 
root@sonic:/home/admin# 
root@sonic:/home/admin# 
root@sonic:/home/admin# 
root@sonic:/home/admin# docker exec -it telemetry ps -ax
  PID TTY      STAT   TIME COMMAND
    1 pts/0    Ss+    0:01 /usr/bin/python /usr/bin/supervisord
   12 pts/0    Sl     0:00 /usr/sbin/rsyslogd -n
   17 pts/0    Sl     0:25 /usr/sbin/telemetry -logtostderr --insecure --port 80
   24 pts/0    Sl     0:18 /usr/sbin/dialout_client_cli -insecure -logtostderr -
   66 pts/1    Rs+    0:00 ps -ax
root@sonic:/home/admin# 
```

**- Description for the changelog**

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

File: rules/redis.mk [hunk 1]
```diff
@@ -13,3 +13,5 @@ REDIS_SENTINEL = redis-sentinel_$(REDIS_VERSION)_amd64.deb
 $(REDIS_SENTINEL)_DEPENDS += $(REDIS_SERVER)
 $(REDIS_SENTINEL)_RDEPENDS += $(REDIS_SERVER)
 $(eval $(call add_derived_package,$(REDIS_TOOLS),$(REDIS_SENTINEL)))
+
+SONIC_STRETCH_DEBS += $(REDIS_TOOLS)
```
Comment (lguohan) [inline rules/redis.mk]: I do not think you need this, the docker target will build this in stretch.

File: rules/telemetry.mk [hunk 2]
```diff
@@ -3,3 +3,5 @@
 SONIC_TELEMETRY = sonic-telemetry_0.1_amd64.deb
 $(SONIC_TELEMETRY)_SRC_PATH = $(SRC_PATH)/telemetry
 SONIC_DPKG_DEBS += $(SONIC_TELEMETRY)
+
+SONIC_STRETCH_DEBS += $(SONIC_TELEMETRY)
```
Comment (lguohan) [inline rules/telemetry.mk]: I do not think you need this, the docker target will build this in stretch.

**Comment by jipanyang** [comment]
One known issue with stretch build is that, even with "make list" or "BLDENV=stretch make list",  it tries to build all missing deb and dockers for strecth.  Guess we could fix that in future.

**Comment by jleveque** [comment]
@jipanyang: Was there a reason not to upgrade `REDIS_VERSION` to the current Stretch default version, `3.2.6-3+deb9u2`?

**Comment by lguohan** [comment]
I do not think so. 3.2.4 still have bugs we should upgrade.

**Comment by jipanyang** [comment]
@jleveque @lguohan  I didn't pay attention to redis version.   Yes, it should be upgraded to stretch default version.  Are there critical bugs with 3.2.4?

**Comment by jipanyang** [comment]
Is going straight to 3.2.13 better?
https://raw.githubusercontent.com/antirez/redis/3.2/00-RELEASENOTES

**Comment by lguohan** [comment]
agree

**Comment by jleveque** [comment]
@jipanyang: As long as the Debian packaging is available for version 3.2.13, we can do it.

Actually, it looks like the current version available from [stretch-backports](https://packages.debian.org/search?suite=stretch-backports&section=all&arch=any&searchon=names&keywords=redis) is 5.0.3-3~bpo9+2. We could potentially upgrade even further, to this version.

**Comment by jipanyang** [comment]
@jleveque  redis is built with packages stored in sonicstorage.blob.core.windows.net https://github.com/Azure/sonic-buildimage/blob/master/src/redis/Makefile#L16,  could you handle the package repo update and version upgrade together?

**Pull request merged**